test.py

```diff
@@ -90,8 +90,14 @@ def path_to(mode, *components):
     build_dir = 'build'
     if os.path.exists(os.path.join(build_dir, 'build.ninja')):
         *dir_components, basename = components
-        return os.path.join(build_dir, *dir_components, all_modes[mode], basename)
-    return os.path.join(build_dir, mode, *components)
+        exe_path = os.path.join(build_dir, *dir_components, all_modes[mode], basename)
+    else:
+        exe_path = os.path.join(build_dir, mode, *components)
+    if not os.access(exe_path, os.F_OK):
+        raise FileNotFoundError(f"{exe_path} is not exist.")
+    elif not os.access(exe_path, os.X_OK):
+        raise PermissionError(f"{exe_path} is not executable.")
```

Comment:
```suggestion
        raise PermissionError(f"{exe_path} is not an executable.")
```

Reply:
or we can check it in a single shot, like
```py
if not os.access(exe_path, os.X_OK | os.F_OK):
    raise PermissionError(f"{exe_path} does not exist or not an executable.")
```

Reply:
Just a nitpick from my side: I think the current error message is correct. Being executable is a property of a file, so `{exe_path} is not executable` sounds right.

Reply:
agreed. i'm withdrawing my suggestion.